[OpenMP] Add a test for D158802

shiltian opened the pull request:
In D158802 we honored user's `thread_limit` value even with the optimization
introduced in D152014. This patch adds a simple test.

jdoerfert reviewed (approved):
Disable it on the CPU or lower the thread count, otherwise this will fail

shiltian committed:
[OpenMP] Add a test for D158802

In D158802 we honored user's `thread_limit` value even with the optimization
introduced in D152014. This patch adds a simple test.
